src/node_i18n.cc

```diff
@@ -62,6 +72,296 @@ bool flag_icu_data_dir = false;
 
 namespace i18n {
 
+Local<Value> ICUException(Environment* env,
+                          UErrorCode status,
+                          const char* msg) {
+  Local<String> estring = OneByteString(env->isolate(), u_errorName(status));
+  if (msg == nullptr || msg[0] == '\0')
+    msg = "Unspecified ICU Exception";
+  Local<String> cons =
+      String::Concat(estring, FIXED_ONE_BYTE_STRING(env->isolate(), ", "));
+  cons = String::Concat(cons, OneByteString(env->isolate(), msg));
```

Comment:
I realize you adapted this code from elsewhere but using snprintf() to format the error message will be much more efficient.